scipy/interpolate/tests/test_polyint.py

```diff
@@ -604,6 +604,23 @@ def test_periodic_eval(self):
         S = CubicSpline(x, y, bc_type='periodic')
         assert_almost_equal(S(1), S(1 + 2 * np.pi), decimal=15)
 
+    def test_second_derivative_continuity_gh_11758(self):
+        # gh-11758: C2 continuity fail
+        x = np.array([0.9, 1.3, 1.9, 2.1, 2.6, 3.0, 3.9, 4.4, 4.7, 5.0, 6.0,
+                      7.0, 8.0, 9.2, 10.5, 11.3, 11.6, 12.0, 12.6, 13.0, 13.3])
+        y = np.array([1.3, 1.5, 1.85, 2.1, 2.6, 2.7, 2.4, 2.15, 2.05, 2.1,
+                      2.25, 2.3, 2.25, 1.95, 1.4, 0.9, 0.7, 0.6, 0.5, 0.4, 1.3])
+        S = CubicSpline(x, y, bc_type='periodic', extrapolate='periodic')
+        self.check_correctness(S, 'periodic', 'periodic')
+
+    def test_three_points(self):
+        # gh-11758: Fails computing a_m2_m1
+        x = np.array([1.0, 2.75, 3.0])
+        y = np.array([1.0, 15.0, 1.0])
+        S = CubicSpline(x, y, bc_type='periodic')
+        self.check_correctness(S, 'periodic', 'periodic')
+        assert_allclose(S.derivative(1)(x), np.array([-48.0, -48.0, -48.0]))
```

Comment:
Why this value? Please add a short comment on how to arrive to this number.

Reply:
In three points case, unknown vector `s` (first derivatives) could be found manually by solving system of 2 linear equations. Due to solution of this system, all `s[i]` are equal and could be computed according to the following formula:

<img src="https://render.githubusercontent.com/render/math?math=s_i=\dfrac{h_1 m_2%2Bh_2m_1}{h_1%2Bh_2} \forall i=0,1,2">

where `h1=x[1]-x[0]`, `h2=x[2]-x[1]`, `m1=y[1]-y[0]/h1`, `m2=y[2]-y[1]/h2`

Reply:
Could add this to explanation as a comment

Reply:
Explanation was added a few lines up.